src/Model/Tag.php

```diff
@@ -70,8 +70,8 @@ class Tag
 	public static function store(int $uriid, int $type, string $name, string $url = '', $probing = true)
 	{
 		if ($type == self::HASHTAG) {
-			// Remove some common "garbarge" from tags
-			$name = trim($name, "\x00..\x20\xFF#!@,;.:'/?!^°$%".'"');
+			// Keep only Unicode word characters
+			$name = preg_replace('/\W/u', '', $name);
 
 			$tags = explode(self::TAG_CHARACTER[self::HASHTAG], $name);
```

Comment:
This function splits the tag if it contains a hashtag - this won't work anymore. So it would be better if that function had been called before this. Also I want to have a look if there are some common non word characters in hashtags.

Reply:
This isn't the place where we determine what is a hashtag or not, we just store it with its name. If there are hashtag names with `#` in them, it's either intended (from protocols with extended entities) or we need to change the way we are looking for hashtags in plain text.

Reply:
Did you figure out which specific non-word characters should be allowed in tag names ?

Reply:
I'm currently heavily involved in family stuff, so my time had been really limited. I will try it these days.

Reply:
It's okay, take your time!